readme.md
# Kletterpartner-Suche
Contributors: Nirus<br />
Tags: Klettern, Climb, Wandern, Walking, Trekking, Climbing, Climb, Walking, Tekking, Hiking, Kletterpartner, Wanderpartner<br />
Requires at least: 4.9.8<br />
Tested up to: 5.6.1<br />
Requires PHP: 5.2.4<br />
Stable tag: 2.4.4<br />
License: GPL Version 3 or later<br />
License URI: https://www.gnu.org/licenses/gpl-3.0.html<br />


The "Climbing-Partner-Search" connects climbing, hiking, tracking and sports fans.

## Installation
1. Unpack and upload the contents of the zipfile to your /wp-content/plugins/ directory or install the plugin through the admin page "Plugins".
2. Activate the plugin via the 'Plugins' screen in WordPress
3. Set the Settings under  Climb-Partner->Settings to configure the plugin
3. Set the GTC and/or GDPR under Climb-Partner->GTC / GDPR
4. Create a page under Menue->Pages
5. Give the page a title.
6. Write the shortcode in the page context: [kps-shortcode]
7. Publish this page.
8.1. Go to the main menu under Design->Menus and add page to a menu.
8.2. Alternatively, you can create a new menu and add the page there
9. Save it. Publish, Done.

###### Tip:
*Of course you can also implement the shortcode in a post.*

## Frequently Asked Questions
If you have questions or problems, use the support forum at Wordpress.org
I will do my best to answer as soon as possible.
[Support-Forum](https://wordpress.org/support/plugin/kletterpartner-suche/)

### I want to translate this plugin
Excellent! Just drop us a message in the support forum and I will register as an editor.
[GlotPress](https://translate.wordpress.org/projects/wp-plugins/kletterpartner-suche/)

### I want to show the form without the entries.
Use the shortcode:
[kps-shortcode show-form-only="true"].

### I want to show the entries without the button.
Use the shortcode:
[kps-shortcode button-write="false"].

### I want to change the form button text.
Use the shortcode:
[kps-shortcode button-text="Your button name"].

### Should I have a GTC for this plugin?
You should decide this yourself.However, I recommend a few ground rules in the GTC set.

### Should I have a GDPR for this plugin?
Yes, you should.
Once personal data is collected, stored or processed, comes the GDPR into play. Inform yourself before there are problems!

### How can I contribute or modify the plugin?
Yes, you can. Under [GitHub](https://github.com/Nirus2000/Kletterpartnersuche) you can actively contribute to the source code, edit bugs and of course bring in improvements.
If you have a Github account, please contact me and I will contact you Add to the committer / contributors, and I look forward to any help in the Support Team.

### What is the GitHub url?
The URL is:
[https://github.com/Nirus2000/Kletterpartnersuche](https://github.com/Nirus2000/Kletterpartnersuche)

## Support
Please use the [support forum.](https://wordpress.org/support/plugin/kletterpartner-suche)
To provide quick support, write down step by step how you came to the problem / error, if necessary add individual partial steps.

If you send me an e-mail, I will not answer.

## Screenshots
1. Form
2. Output
3. Settings in the admin panel
4. Settings in the admin panel
5. Settings in the admin panel
6. Settings in the admin panel
7. Simple integration of the GTC or GDPR
8. Settings for the email templates
9. Settings for different icons, widget and other things
10. Settings for translations of the form
11. Overview of entries in the admin panel

## Description
The "Climbing-Partner-Search" connects climbing, hiking, tracking and sports fans. It was for the Saxon Mountaineering Federation e. V.
(a section of the DAV - German Alpine Association e.V.) developed.

### What can this plugin do?
* Simple installation with shortcodes
* Automatic sending of emails for activation and verification
* Editable email templates for every action
* Users can delete their own entries
* Report entries
* Integration of own GTC / GDPR
* Widget
* IconPaks in diffrend colors
* Options in the form freely selectable

### Which sports are available for selection?
* Climb
* Bouldering
* Ferratas
* Alpine tours
* Hiking
* Trekking
* Travel
* Indoor sports
* Mountain bike
* Kayak
* Ski / Snowboarding
* Winter sports

## Upgrade Notice
Bugfix and extension of the plugin...

## Changelog
= v. 2.4.4 =
*[18.02.2021]*
* Fix: ReadMe.md
* Fix: Some translations

## Changelog
= v. 2.4.3 =
*[24.11.2020]*
* Fix: Toggle in Adminpanel

= v. 2.4.2 =
*[13.07.2020]*
* Fix: Activation-Link

= v. 2.4.1 =
*[09.06.2020]*
* Fix: DropDown menu "Please Choose" in form

= v. 2.4.0 =
*[18.04.2020]*
* Add: Sports kayak, bouldering, ferratas, alpine tours, hiking
* Add: Some reset button in email templates
* Add: Reset button for statistics
* Add: Reset button for translations
* Fix: Some small HTML bugs
* Fix: Some small PHP bugs
* Upd: FontAwesome to 5.13.0

= v. 2.3.1 =
*[14.10.2019]*
* Fix: Install

= v. 2.3.0 =
*[13.10.2019]*
* Add: Choose form options
* Add: Design->Translations
* Fix: IconPaks Orange
* Fix: Autor Avatar
* Fix: Email templates

= v. 2.2.3 =
*[04.10.2019]*
* Fix: Some small HTML bugs
* Fix: Some small PHP bugs

= v. 2.2.2 =
*[15.09.2019]*
* Fix: IconPaks for Android

= v. 2.2.1 =
*[12.09.2019]*
* Fix: Activation-Email

= v. 2.2.0 =
*[03.09.2019]*
* Upd: Optimize source code

= v. 2.1.1 =
*[30.08.2019]*
* Add: changelog.txt
* Fix: Install
* Upd: readme.txt
* Upd. readme.md

= v. 2.1 =
*[29.08.2019]*
* Add: Some IconPak
* Add: Design->Output Legend activation
* Fix: Widget
* Upd: Optimize source code
* Upd: IconPak
* Upd: Clear readme.txt
* Upd: Clear readme.md
* Del: Design->Legend Settings
* Del: Design->Widget Settings

= v. 2.0 =
*[16.06.2019]*
* Add: Activation link for administrator
* Upd: Optimize source code

= v.1.0 to 1.9 =

Look at changelog.txt

changelog.txt
= v. 2.4.4 =
*[18.02.2021]*
* Fix: ReadMe.md
* Fix: Some translations

= v. 2.4.3 =
*[24.11.2020]*
* Fix: Toggle in Adminpanel

= v. 2.4.2 =
*[13.07.2020]*
* Fix: Activation-Link

= v. 2.4.1 =
*[09.06.2020]*
* Fix: DropDown menu "Please Choose" in form

= v. 2.4.0 =
*[15.11.2019]*
* Upd: Translation de_DE
* Upd: Translation de_DE_formal
* Upd: FontAwesome to 5.11.2
* Upd: readme.txt
* Upd. readme.md

= v. 2.3.1 =
*[14.10.2019]*
* Fix: Install

= v. 2.3.0 =
*[13.10.2019]*
* Add: Choose form options
* Add: Design->Translations
* Fix: IconPaks Orange
* Fix: Autor Avatar
* Fix: Email templates

= v. 2.2.3 =
*[04.10.2019]*
* Fix: Some small HTML bugs
* Fix: Some small PHP bugs

= v. 2.2.2 =
*[15.09.2019]*
* Fix: IconPaks for Android

= v. 2.2.1 =
*[12.09.2019]*
* Fix: Activation-Email

= v. 2.2.0 =
*[03.09.2019]*
* Upd: Optimize source code

= v. 2.1.1 =
*[30.08.2019]*
* Upd: readme.txt
* Upd. readme.md
* Fix: Install
* Add: changelog.txt

= v. 2.1 =
*[29.08.2019]*
* Add: Some IconPak
* Add: Design->Output Legend activation
* Upd: Optimize source code
* Upd: IconPak
* Upd: Clear readme.txt
* Upd: Clear readme.md
* Fix: Widget
* Del: Design->Legend Settings
* Del: Design->Widget Settings

= v. 2.0 =
*[16.06.2019]*
* Add: Activation link for administrator
* Upd: Optimize source code

= v. 1.9 =
*[27.04.2019]*
* Add: Form option Messenger Threema
* Upd: Optimize source code
* Upd: Some checks of variables
* Upd: Pagination Backend
* Upd: Pagination Frontend
* Upd: FontAwesome to 5.8.1
* Fix: Form escape

= v. 1.81 =
*[12.04.2019]*
* Fix: Output entries
* Upd: PHP requirements to 5.2.4

= v. 1.8 =
*[10.04.2019]*
* Add: Shorttag for change form button text
* Fix: Pagination admin panel
* Fix: Install update function
* Fix: Install
* Fix: Direct link in Menue->Plugins
* Fix: Released time translation in admin panel
* Fix: Form escape
* Upd: Translation de_DE
* Upd: Translation de_DE_formal
* Upd: Translation en_US

= v. 1.72 =
*[19.02.2019]*
* Upd: GPL Version 3.0
* Upd: Readme file
* Upd: Translation de_DE
* Upd: Translation de_DE_formal
* Upd: Translation en_US

= v. 1.71 =
*[18.02.2019]*
* Upd: Readme file
* Upd: Translation de_DE
* Upd: Translation de_DE_formal

= v. 1.7 =
*[16.02.2019]*
* Add: Design Time format in date output
* Fix: CSS-Style entry button
* Fix: Setting e-mail address
* Fix: Shorttag examples in e-mail templates
* Fix: Activation email release time
* Fix: Translation (Singular/Plural)
* Fix: Missing/false translation
* Fix: Form Close-SVG
* Fix: E-mail header information
* Fix: E-mail copy information
* Upd: Checkbox Background-Color
* Upd: FontAwesome to 5.7.2
* Upd: DashIcon in form
* Upd: Glotpress source code translation in english

= v. 1.6 =
*[21.11.2018]*
* Add: Design Shortcode overview
* Add: GitHub URL
* Fix: Setting overview (short)
* Fix: Required fields in form
* Fix: Verification
* Fix: Install
* Fix: Database
* Upd: FAQs

= v. 1.5 =
*[20.11.2018]*
* Fix: Distance option fields
* Fix: Readme file

= v. 1.4 =
*[14.10.2018]*
* Add: DashIcon in form
* Add: Form option Messenger Viper
* Add: Form option Messenger Signal
* Add: Form option Messenger Telegram
* Add: Form option Messenger Whatsapp
* Add: Form option Messenger Facebook
* Add: Form option Messenger Hoccer
* Add: Form option Messenger Skype
* Add: Form option Messenger Wire
* Add: Form option Website
* Fix: Widget no entry
* Fix: Form error entry existing
* Fix: Form error entry insert in database
* Fix: Javascript loading in header
* Fix: HTML5 Input-Type "tel"
* Fix: Send e-mail, if entry was released
* Fix: Legend

= v. 1.3 =
*[11.10.2018]*
* Add: Widget
* Add: Export for personal data
* Add: Erasure for personal data
* Fix: Additional information via e-mail
* Fix: Form option
* Upd: GTC / GDPR status
* Upd: Activation
* Del: Form option Google+ (Closed Aug. 2019)


= v. 1.2 =
*[05.10.2018]*
* Add: Quicklaunch "locked by message"
* Add: Quicklaunch "Auto-Lock"
* Add: Reports are logged
* Add: Settings -> Reports
* Add: Reporting
* Add: Design
* Add: Icon-Pak
* Add: Legend
* Add: Legende-Pak
* Add: Tabs in the Settings
* Fix: Link "Edit" for moderators
* Fix: Show Author-IP
* Fix: Form checkbox "GTC/GDPR"
* Fix: Test checkbox "GTC/GDPR"
* Fix: Shorttage %linkactivation%
* Fix: Shorttage %linkdelete%
* Fix: Form errors
* Fix: Small translation errors
* Upd: Complet redesign in the admin panel
* Upd: div. Variablen umbenannt
* Upd: CSS


= v. 1.1 =
*[27.09.2018]*
* Fix: Permissions in the admin panel
* Fix: Page makeup options labels in the form
* Fix: Uninstall only by role adminnistrator


= v. 1.0 =
* 1st Release

admin/kps-install.php
<?php
/**
 * @author        Alexander Ott
 * @copyright     2018-2021
 * @email         kps@nirus-online.de
 *
 * All rights reserved
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.

 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.

 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.

 * Dieses Programm ist Freie Software: Sie können es unter den Bedingungen
 * der GNU General Public License, wie von der Free Software Foundation,
 * Version 3 der Lizenz oder (nach Ihrer Wahl) jeder neueren
 * veröffentlichten Version, weiterverbreiten und/oder modifizieren.

 * Dieses Programm wird in der Hoffnung, dass es nützlich sein wird, aber
 * OHNE JEDE GEWÄHRLEISTUNG, bereitgestellt; sogar ohne die implizite
 * Gewährleistung der MARKTFÄHIGKEIT oder EIGNUNG FÜR EINEN BESTIMMTEN ZWECK.
 * Siehe die GNU General Public License für weitere Details.

 * Sie sollten eine Kopie der GNU General Public License zusammen mit diesem
 * Programm erhalten haben. Wenn nicht, siehe <http://www.gnu.org/licenses/>.
 */

// Kein direkten Zugriff erlauben
if (strpos($_SERVER['PHP_SELF'], basename(__FILE__)))
{
    wp_die(__('No direct calls allowed!'));
}

/**
 * Installationsfunktion
 */
function kps_install()
{
    global $wpdb;

    // Tabelle existiert nicht, dann erstellen
    $result_entries         = $wpdb->query("SHOW TABLES LIKE '" . $wpdb->prefix . "kps_entries'");
    $result_requirements    = $wpdb->query("SHOW TABLES LIKE '" . $wpdb->prefix . "kps_requirement'");

    if ($result_entries == 0)
    {
        $sql = "
                CREATE TABLE
                " . KPS_TABLE_ENTRIES . "
                (
                id int(11) NOT NULL auto_increment,
                authorName varchar(100) NOT NULL,
                authorId int(11) NOT NULL,
                authorEmail varchar(255) NOT NULL,
                password varchar(64) NOT NULL,
                activationHash varchar(32) NOT NULL,
                deleteHash varchar(32) NOT NULL,
                hash varchar(64) NOT NULL,
                content longtext NOT NULL,
                setDateTime bigint(8) UNSIGNED NOT NULL,
                unlockDateTime bigint(8) UNSIGNED NOT NULL,
                deleteDateTime bigint(8) UNSIGNED NOT NULL,
                authorSearchfor tinyint(1) NOT NULL,
                authorRule tinyint(1) NOT NULL,
                yourRule tinyint(1) NOT NULL,
                formOptions mediumtext NOT NULL,
                isLocked tinyint(1) NOT NULL,
                isLockedByAdmin tinyint(1) NOT NULL,
                lockedAutoReport TINYINT(1) NOT NULL,
                authorIp mediumtext NOT NULL,
                authorHost mediumtext NOT NULL,
                reportCount mediumtext NOT NULL,
                isReported tinyint(1) NOT NULL,
                PRIMARY KEY  (id)
                ) ENGINE=MyISAM CHARACTER SET utf8 COLLATE utf8_general_ci";
        $result_entries = $wpdb->query($sql);
    }

    if ($result_requirements == 0)
    {
        $sql = "
                CREATE TABLE
                " . KPS_TABLE_REQUIREMENT . "
                (
                id int(11) NOT NULL auto_increment,
                entryId int(11) NOT NULL,
                password varchar(64) NOT NULL,
                hash varchar(64) NOT NULL,
                userEmail varchar(255) NOT NULL,
                timestamp bigint(8) UNSIGNED NOT NULL,
                expire bigint(8) UNSIGNED NOT NULL,
                sendTimestamp BIGINT(8) UNSIGNED NOT NULL,
                sendData tinyint(1) NOT NULL,
                PRIMARY KEY  (id)
                ) ENGINE=MyISAM CHARACTER SET utf8 COLLATE utf8_general_ci";
        $result_requirements = $wpdb->query($sql);
    }

    // Warte 1 Sekunde für Delay
    sleep(1);

    // Prüfen, ob Tabelle existiert nach Installation
    $resultkpsEntriesNotExist       = $wpdb->query("SHOW TABLES LIKE '" . $wpdb->prefix . "kps_entries'");
    $resultkpsRequirementNotExist   = $wpdb->query("SHOW TABLES LIKE '" . $wpdb->prefix . "kps_requirement'");

    if ($resultkpsEntriesNotExist != 0 && $resultkpsRequirementNotExist !=0)
    {
        // Hole Wordpress Haupt-Email-Adresse
        $sendEmail = get_bloginfo('admin_email', 'raw');

        // Setze Wordpress Haupt-Email-Adresse für Grundeinstellung
        $sendEmailCC = serialize(array(
            'kpsEmailCC'                        => get_bloginfo('admin_email', 'raw'),
            'kpsEmailReport'                    => get_bloginfo('admin_email', 'raw'),
            'kpsEmailInformation'               => 'false'
        ));

        // FormularOptionen
        $formOptions = serialize(array(
            'kpsFormOptionHall'                 => 'true',
            'kpsFormOptionClimbing'             => 'true',
            'kpsFormOptionWalking'              => 'true',
            'kpsFormOptionAlpineTours'          => 'true',
            'kpsFormOptionKayak'                => 'true',
            'kpsFormOptionFerratas'             => 'true',
            'kpsFormOptionMountainBike'         => 'true',
            'kpsFormOptionWinterSports'         => 'true',
            'kpsFormOptionTravels'              => 'true',
            'kpsFormOptionOneTime'              => 'true',
            'kpsFormOptionMoreTime'             => 'true',
            'kpsFormOptionSinglePerson'         => 'true',
            'kpsFormOptionFamily'               => 'true',
            'kpsFormOptionClubGroup'            => 'true',
            'kpsFormOptionTelephone'            => 'true',
            'kpsFormOptionMobile'               => 'true',
            'kpsFormOptionWhatsapp'             => 'true',
            'kpsFormOptionSignal'               => 'true',
            'kpsFormOptionViper'                => 'false',
            'kpsFormOptionTelegram'             => 'true',
            'kpsFormOptionThreema'              => 'true',
            'kpsFormOptionFacebookMessenger'    => 'false',
            'kpsFormOptionWire'                 => 'false',
            'kpsFormOptionHoccer'               => 'false',
            'kpsFormOptionSkype'                => 'false',
            'kpsFormOptionWebsite'              => 'true',
            'kpsFormOptionFacebook'             => 'false',
            'kpsFormOptionInstagram'            => 'false'
        ));

        // FormularOptionen Users
        $userSettings = serialize(array(
            'kpsUserRequireRegistration'        => 'false',
            'kpsUserRequirementRegistration'    => 'false',
            'kpsUserRequireAdminUnlock'         => 'false',
            'kpsUserProfilLink'                 => 'false',
            'kpsUserReport'                     => 'false',
            'kpsUserPrivacyAGB'                 => 'false',
            'kpsUserPrivacyDSGVO'               => 'false',
            'kpsUserAvatar'                     => 'false',
            'kpsUserRequirementReport'          => 'false',
            'kpsUserReport'                     => 'false'
        ));

        // Einträge melden
        $report = serialize(array(
            'kpsReportActivation'               => 'false',
            'kpsAdminSendReportAfter'           => 70,
            'kpsReportSpam'                     => 25,
            'kpsAutoReportSpam'                 => 50,
            'kpsReportUnreasonable'             => 25,
            'kpsAutoReportUnreasonable'         => 50,
            'kpsReportDouble'                   => 25,
            'kpsAutoReportDouble'               => 50,
            'kpsReportPrivacy'                  => 25,
            'kpsAutoReportPrivacy'              => 50,
            'kpsReportOthers'                   => 25,
            'kpsAutoReportOthers'               => 50
        ));

        // KPS-Counter
        $kpsCounter = serialize(array(
            'kpsAllEntrys'                      => 0,
            'kpsAllActivatedEntrys'             => 0,
            'kpsAllVerfifications'              => 0,
            'kpsAllSendRequirements'            => 0,
            'kpsAllDeleteEntrys'                => 0
        ));

        // Output
        $output = serialize(array(
            'kpsUnlockTime'                     => 'false',
            'kpsEmailSetTime'                   => 'true',
            'kpsEmailUnlockTime'                => 'true',
            'kpsEmailDeleteTime'                => 'true',
            'kpsLegendActivated'                => 'true'
        ));

        // Einstellungen speichern in der Options-Tabelle
        add_option('kps_version', KPS_VER);                             // KPS-Version
        add_option('kps_formOptions', $formOptions);                    // Formularoptionen
        add_option('kps_formWordCount', 25);                            // mindestwörter im Formular
        add_option('kps_captchakeys', '');                              // Captcha-Keys
        add_option('kps_captcha', 'false');                             // Captcha
        add_option('kps_userSettings', $userSettings);                  // Registierung Pflicht für Einträge
        add_option('kps_backendPagination', 10);                        // Einträge pro Seite im Backend
        add_option('kps_frontendPagination', 5);                        // Einträge pro Seite im Frontend
        add_option('kps_mailFrom', $sendEmail);                         // Email-Versendungsadresse
        add_option('kps_mailFromCC', $sendEmailCC);                     // Email-Versendungsadresse Kopie
        add_option('kps_deleteEntryTime', 7776000);                     // Verfallszeit eines Eintrages -> entspricht 90 Tagen
        add_option('kps_deleteNoEntryTime', 5184000);                   // Verfallszeit eines nicht freigeschaltenen Eintrages -> entspricht 60 Tagen
        add_option('kps_agb', 0);                                       // AGB-Seite
        add_option('kps_dsgvo', 0);                                     // DSGVO-Seite
        add_option('kps_report', $report);                              // Einträge melden
        add_option('kps_icon', 17);                                     // FormOptions / Widget IconPak
        add_option('kps_kpsCounter', $kpsCounter);                      // KPS-Counter
        add_option('kps_output', $output);                              // Ausgabeneinstellungen
        add_option('kps_translations', '');                             // Übersetzungen
    }
    else
    {
        wp_die(__('Databases already exist. Installation aborted. Contact your administrator!'));
    }
}

/**
 * Updatefunktion
 * Update von früheren Versionen des Plugins
 */
function kps_upgrade()
{
    global $wpdb;

    // Derzeitige Version holen
    $current_version = get_option('kps_version');

    if (version_compare($current_version, '1.1', '<'))
    {
        wp_die(__('This version is discontinued. Please uninstall and delete this plugin. A new version is available!'));
        exit;
    }

    if (version_compare($current_version, '1.2', '<'))
    {
        // Versendungszeitstempel einfügen
        $sql =  "ALTER TABLE
                " . KPS_TABLE_REQUIREMENT . "
                ADD sendTimestamp BIGINT(8) NOT NULL AFTER expire";
        $result = $wpdb->query($sql);

        // Widget anzeigen
        $widget = serialize(array(
            'kpsWidgetIconPak'  => 0
        ));

        // Widget updaten
        update_option('kps_widget', $widget, 'yes');
    }

    if (version_compare($current_version, '1.3', '<'))
    {
        // Formular-Optionen updaten
        $formOptions = kps_unserialize(get_option('kps_formOptions'));

        $setFormOption['kpsFormOptionTelephone']            = $formOptions['kpsFormOptionTelephone'];
        $setFormOption['kpsFormOptionMobile']               = $formOptions['kpsFormOptionMobile'];
        $setFormOption['kpsFormOptionWhatsapp']             = 'false';
        $setFormOption['kpsFormOptionSignal']               = 'false';
        $setFormOption['kpsFormOptionViper']                = 'false';
        $setFormOption['kpsFormOptionTelegram']             = 'false';
        $setFormOption['kpsFormOptionFacebookMessenger']    = 'false';
        $setFormOption['kpsFormOptionWire']                 = 'false';
        $setFormOption['kpsFormOptionHoccer']               = 'false';
        $setFormOption['kpsFormOptionSkype']                = 'false';
        $setFormOption['kpsFormOptionWebsite']              = 'false';
        $setFormOption['kpsFormOptionInstagram']            = $formOptions['kpsFormOptionInstagram'];
        $setFormOption['kpsFormOptionFacebook']             = $formOptions['kpsFormOptionFacebook'];

        $setFormOption = serialize($setFormOption);

        // Formularoptionen updaten
        update_option('kps_formOptions', $setFormOption);
    }

    if (version_compare($current_version, '1.5', '<'))
    {
        $output = serialize(array(
            'kpsUnlockTime'                                 => 'false',
            'kpsEmailSetTime'                               => 'true',
            'kpsEmailUnlockTime'                            => 'true',
            'kpsEmailDeleteTime'                            => 'true'
        ));

        // Erstelle Ausgabe
        add_option('kps_output', $output);
    }

    if (version_compare($current_version, '1.81', '<'))
    {
        // Formular-Optionen updaten
        $formOptions = kps_unserialize(get_option('kps_formOptions'));

        $setFormOption['kpsFormOptionTelephone']            = $formOptions['kpsFormOptionTelephone'];
        $setFormOption['kpsFormOptionMobile']               = $formOptions['kpsFormOptionMobile'];
        $setFormOption['kpsFormOptionWhatsapp']             = $formOptions['kpsFormOptionWhatsapp'];
        $setFormOption['kpsFormOptionSignal']               = $formOptions['kpsFormOptionSignal'];
        $setFormOption['kpsFormOptionViper']                = $formOptions['kpsFormOptionViper'];
        $setFormOption['kpsFormOptionTelegram']             = $formOptions['kpsFormOptionTelegram'];
        $setFormOption['kpsFormOptionThreema']              = 'false';
        $setFormOption['kpsFormOptionFacebookMessenger']    = $formOptions['kpsFormOptionFacebookMessenger'];
        $setFormOption['kpsFormOptionWire']                 = $formOptions['kpsFormOptionWire'];
        $setFormOption['kpsFormOptionHoccer']               = $formOptions['kpsFormOptionHoccer'];
        $setFormOption['kpsFormOptionSkype']                = $formOptions['kpsFormOptionSkype'];
        $setFormOption['kpsFormOptionWebsite']              = $formOptions['kpsFormOptionWebsite'];
        $setFormOption['kpsFormOptionInstagram']            = $formOptions['kpsFormOptionInstagram'];
        $setFormOption['kpsFormOptionFacebook']             = $formOptions['kpsFormOptionFacebook'];

        $setFormOption = serialize($setFormOption);

        // Formularoptionen updaten
        update_option('kps_formOptions', $setFormOption);
    }

    if (version_compare($current_version, '2.1', '<'))
    {
        // Legende-Optionen auslesen
        $legend             = kps_unserialize(get_option('kps_legend'));
        $legendActivated    = ($legend['kpsLegendActivated'] === 'true') ? 'true' : 'false';

        // Output-Optionen updaten
        $output = kps_unserialize(get_option('kps_output'));

        $setOutput['kpsUnlockTime']                         = $output['kpsUnlockTime'];
        $setOutput['kpsEmailSetTime']                       = $output['kpsEmailSetTime'];
        $setOutput['kpsEmailUnlockTime']                    = $output['kpsEmailUnlockTime'];
        $setOutput['kpsEmailDeleteTime']                    = $output['kpsEmailDeleteTime'];
        $setOutput['kpsLegendActivated']                    = $legendActivated;

        $setOutput = serialize($setOutput);

        // Formularoptionen updaten
        update_option('kps_output', $setOutput);

        // IconPak updaten
        update_option('kps_icon', 17);

        // Lösche Option Legende
        delete_option('kps_legend');

        // Lösche Option Widget
        delete_option('kps_widget');
    }

    if (version_compare($current_version, '2.3.1', '<'))
    {
        // Formular-Optionen updaten
        $setFormOption['kpsFormOptionHall']                 = 'true';
        $setFormOption['kpsFormOptionClimbing']             = 'true';
        $setFormOption['kpsFormOptionWalking']              = 'true';
        $setFormOption['kpsFormOptionTravels']              = 'true';
        $setFormOption['kpsFormOptionSinglePerson']         = 'true';
        $setFormOption['kpsFormOptionFamily']               = 'true';
        $setFormOption['kpsFormOptionClubGroup']            = 'true';
        $setFormOption['kpsFormOptionTelephone']            = 'true';
        $setFormOption['kpsFormOptionMobile']               = 'true';
        $setFormOption['kpsFormOptionWhatsapp']             = 'true';
        $setFormOption['kpsFormOptionSignal']               = 'true';
        $setFormOption['kpsFormOptionViper']                = 'false';
        $setFormOption['kpsFormOptionTelegram']             = 'true';
        $setFormOption['kpsFormOptionThreema']              = 'true';
        $setFormOption['kpsFormOptionFacebookMessenger']    = 'false';
        $setFormOption['kpsFormOptionWire']                 = 'false';
        $setFormOption['kpsFormOptionHoccer']               = 'false';
        $setFormOption['kpsFormOptionSkype']                = 'false';
        $setFormOption['kpsFormOptionWebsite']              = 'true';
        $setFormOption['kpsFormOptionInstagram']            = 'false';
        $setFormOption['kpsFormOptionFacebook']             = 'false';

        $setFormOption = serialize($setFormOption);

        // Formularoptionen updaten
        update_option('kps_formOptions', $setFormOption);

        // Erstelle Übersetzungen
        add_option('kps_translations', '');                             // Übersetzungen
    }

    if (version_compare($current_version, '2.4.0', '<'))
    {

        // Formular-Optionen updaten
        $formOptions = kps_unserialize(get_option('kps_formOptions'));

        // Formular-Optionen updaten
        $setFormOption['kpsFormOptionHall']                 = 'true';
        $setFormOption['kpsFormOptionClimbing']             = 'true';
        $setFormOption['kpsFormOptionWalking']              = 'true';
        $setFormOption['kpsFormOptionAlpineTours']          = 'true';
        $setFormOption['kpsFormOptionKayak']                = 'true';
        $setFormOption['kpsFormOptionFerratas']             = 'true';
        $setFormOption['kpsFormOptionMountainBike']         = 'true';
        $setFormOption['kpsFormOptionWinterSports']         = 'true';
        $setFormOption['kpsFormOptionKayak']                = 'true';
        $setFormOption['kpsFormOptionTravels']              = 'true';
        $setFormOption['kpsFormOptionOneTime']              = 'true';
        $setFormOption['kpsFormOptionMoreTime']             = 'true';
        $setFormOption['kpsFormOptionSinglePerson']         = 'true';
        $setFormOption['kpsFormOptionFamily']               = 'true';
        $setFormOption['kpsFormOptionClubGroup']            = 'true';
        $setFormOption['kpsFormOptionTelephone']            = $formOptions['kpsFormOptionTelephone'];
        $setFormOption['kpsFormOptionMobile']               = $formOptions['kpsFormOptionMobile'];
        $setFormOption['kpsFormOptionWhatsapp']             = $formOptions['kpsFormOptionWhatsapp'];
        $setFormOption['kpsFormOptionSignal']               = $formOptions['kpsFormOptionSignal'];
        $setFormOption['kpsFormOptionViper']                = $formOptions['kpsFormOptionViper'];
        $setFormOption['kpsFormOptionTelegram']             = $formOptions['kpsFormOptionTelegram'];
        $setFormOption['kpsFormOptionThreema']              = $formOptions['kpsFormOptionThreema'];
        $setFormOption['kpsFormOptionFacebookMessenger']    = $formOptions['kpsFormOptionFacebookMessenger'];
        $setFormOption['kpsFormOptionWire']                 = $formOptions['kpsFormOptionWire'];
        $setFormOption['kpsFormOptionHoccer']               = $formOptions['kpsFormOptionHoccer'];
        $setFormOption['kpsFormOptionSkype']                = $formOptions['kpsFormOptionSkype'];
        $setFormOption['kpsFormOptionWebsite']              = $formOptions['kpsFormOptionWebsite'];
        $setFormOption['kpsFormOptionInstagram']            = $formOptions['kpsFormOptionInstagram'];
        $setFormOption['kpsFormOptionFacebook']             = $formOptions['kpsFormOptionFacebook'];

        $setFormOption = serialize($setFormOption);

        // Formularoptionen updaten
        update_option('kps_formOptions', $setFormOption);

        // Übersetzungen updaten
        delete_option('kps_translations');
        add_option('kps_translations', '');                          // Übersetzungen
    }

    if (version_compare($current_version, '2.4.2', '<'))
    {
        // Lösche Option Emailsettings
        delete_option('kps_authorMailSettings');
    }

    // Version updaten
    update_option('kps_version', KPS_VER);
}

